plugins/nextcloud-rule-exclusions-before.conf

```diff
@@ -792,6 +843,63 @@ SecRule REQUEST_FILENAME "@rx /remote\.php/dav/calendars/[^/]+/trashbin/objects/
         "t:none,\
         ctl:ruleRemoveById=200002"
 
+# Creating new calendar groups in web GUI
+# The / at the end is sometimes not included
+SecRule REQUEST_FILENAME "@rx ^/remote\.php/dav/calendars/[^/]+/[^/]+/?$" \
+    "id:9508339,\
+    phase:1,\
+    pass,\
+    t:none,\
+    nolog,\
+    ver:'nextcloud-rule-exclusions-plugin/1.1.0',\
+    chain"
+    SecRule REQUEST_METHOD "@pm REPORT MKCOL" \
+        "t:none,\
+        ctl:ruleRemoveTargetById=920272;REQUEST_BODY,\
+        ctl:ruleRemoveTargetById=920273;REQUEST_BODY,\
+        ctl:ruleRemoveTargetById=932236;XML:/*,\
+        ctl:ruleRemoveTargetById=942430;XML:/*,\
+        ctl:ruleRemoveTargetById=942431;XML:/*,\
+        ctl:ruleRemoveTargetById=942432;XML:/*,\
+        ctl:ruleRemoveTargetById=942440;XML:/*"
+
+# Enabling/disabling "Enable birthday calendar setting
+# Deleting an calendar group
+# The / at the end is sometimes not included
+# XML parser fails because there is no request body
+SecRule REQUEST_FILENAME "@rx /remote\.php/dav/calendars/[^/]+/[^/]+/?$" \
+    "id:9508360,\
+    phase:1,\
+    pass,\
+    t:none,\
+    nolog,\
+    ver:'nextcloud-rule-exclusions-plugin/1.1.0',\
+    chain"
+    SecRule REQUEST_METHOD "@pm DELETE" \
+        "t:none,\
+        chain"
+        SecRule &REQUEST_BODY_LENGTH "@eq 1" \
```

Comment:
I'm not sure this is a valid expression, I mean `&REQUEST_BODY_LENGTH` makes no sense. `REQUEST_BODY_LENGTH` provides the length of request body - but what could mean `&` as a prefix?

And looks like Apache does not allow this, see the CI [error](https://github.com/coreruleset/nextcloud-rule-exclusions-plugin/actions/runs/8529308350/job/23364805222?pr=62#step:5:12).

Reply:
I was thinking it meant 0 bytes in the request body, but I now realize that could also mean the number of request bodies in a request(afaik you can't have more than 1 request body, I'm not sure how that would even work). An example isn't provided [in the docs](https://github.com/owasp-modsecurity/ModSecurity/wiki/Reference-Manual-(v3.x)#user-content-REQUEST_BODY_LENGTH) so that's probably part of my confusion. I think the correct value here is ``REQUEST_BODY_LENGTH "@eq 0"`` but I'll be testing to make sure.

Reply:
As I see libmodsecurity3 allows this syntax, and the value (of `&REQUEST_BODY_LENGTH`) will be evaluated as `1` (even if the value of `REQUEST_BODY_LENGTH` is for eg. `12`).

I suggest you use `&REQUEST_BODY` or `REQUEST_BODY_LENGTH`, but not `&REQUEST_BODY_LENGTH`.

Note: it's not relevant here but the Apache's error message is a bit misleading: `Error creating rule: The & modificator does not apply to non-collection variables.`, because `REQUEST_BODY` is not a collection, but it allows to use `&` with that.

Reply:
``REQUEST_BODY_LENGTH "@eq 0"`` does it, everything should be fixed now. Thanks for clearing up my confusion.